### ISSUE TITLE
demos(select): select header example is not accessible

### ISSUE THREAD

**Issue by Splaktar** [angular/material#11139]
### Bug, feature request, or proposal:
As mentioned in https://github.com/angular/material/issues/10342#issuecomment-308556638, the current `md-select-header` demo is not accessible and will lead developers to implement inaccessible select search boxes in their apps.

### What is the expected behavior?
All of the demos should be using accessibility best practices and be considered 'accessible'.

### What is the current behavior?
Currently the demo just sticks an `input` in the `md-select-header`. While that `input` has a `placeholder`, it lacks a number of the accessibility features of `md-input`. There may also be some accessibility issues in the general interaction with the component and initial focus state as described in https://github.com/angular/material/issues/9146.

##### Detailed Reproduction Steps: 
Test with screen reader.

### What is the use-case or motivation for changing an existing behavior?
To align with the accessibility goals of this project.

### Which versions of AngularJS, Material, OS, and browsers are affected?
AngularJS 1.1.7

### Is there anything else we should know? Stack Traces, Screenshots, etc.
No
